sdk/python/feast/infra/provider.py

```diff
@@ -144,7 +146,30 @@ def get_provider(config: RepoConfig, repo_path: Path) -> Provider:
 
         return LocalProvider(config, repo_path)
     else:
-        raise ValueError(config)
+        if "." not in config.provider:
+            raise errors.ProviderNameParsingException(config.provider)
+        # Split provider into module and class names by finding the right-most dot.
+        # For example, provider 'foo.bar.MyProvider' will be parsed into 'foo.bar' and 'MyProvider'
+        module_name, class_name = config.provider.rsplit(".", 1)
+
+        # Try importing the module that contains the custom provider
+        try:
+            module = importlib.import_module(module_name)
+        except Exception as e:
+            # The original exception can be anything - either module not found,
+            # or any other kind of error happening during the module import time.
+            # So we should include the original error as well in the stack trace.
+            raise errors.ProviderModuleImportError(module_name) from e
+
+        # Try getting the provider class definition
+        try:
+            ProviderCls = getattr(module, class_name)
+        except AttributeError:
+            # This can only be one type of error, when class_name attribute does not exist in the module
+            # So we don't have to include the original exception here
+            raise errors.ProviderClassImportError(module_name, class_name) from None
+
+        return ProviderCls(config)
```

Comment:
We should probably pass in the `repo_path` for now, as with the `LocalProvider`. It's used for relative imports in some cases. It should definitely be cleaned up, but probably not as part of this PR.